### PR TITLE
Add cmake options to handle logging

### PR DESCRIPTION
**Pull request by Yaraslaut**
Introduce compile definitions to handle debug output for endo

### CODE REVIEW

File: src/shell/Shell.cpp [hunk 1]
```diff
@@ -645,8 +641,13 @@ export class Shell final: public CoreVM::Runtime
     std::unique_ptr<CoreVM::Program> _currentProgram;
     CoreVM::Runner::Globals _globals;
 
+#if defined(ENDO_TRACE)
```
Comment (christianparpart) [inline src/shell/Shell.cpp:644]: I think that should be `ENDO_TRACE_VM` then. What do you think? Reason is, I was confused about `ENDO_TRACE`'s use, when ._PARSER and ._LEXER were pretty obvious to me.

On the other hand, what about having one trace preprocessor define, and then enable it fine grained at runtime, similar to logstore in Contour?

Comment (Yaraslaut) [reply src/shell/Shell.cpp:644]: Issue with runtime would be to enable this logs for the tests

File: CMakeLists.txt [hunk 2]
```diff
@@ -35,3 +45,10 @@ enable_testing()
 
 add_subdirectory(src)
 EndoThirdPartiesSummary2()
+message(STATUS "--------------------------------")
+message(STATUS "          output flags          ")
+message(STATUS "--------------------------------")
+message(STATUS "endo trace          ${ENDO_TRACE_VM}")
```
Comment (christianparpart) [inline CMakeLists.txt:51]: Change String on left side then too? :)

Comment (Yaraslaut) [reply CMakeLists.txt]: Changed

**Review by christianparpart — APPROVED** [review]
:-)

**Pull request merged**